### DISCUSSION TITLE
How to handling polling and rendering stale data

### DISCUSSION THREAD

**Discussion by RobinClowers** [facebookexperimental/Recoil#2087] [Q&A] [answered]
It would be great if there way a way to keep stale data around until an async selector refreshes. Right now I can write a selector that fetches data, and use `useRecoilRefresher_UNSTABLE` to poll on an interval, but every time it polls, I see the suspense fallback in my UI, which is not ideal.

As far as I can see, the only way around this is to write a hook that fetches and writes data to an atom on success. This is fine for simple cases, but if I have to make a few requests in sequence to get the data, it gets pretty complicated.

Any suggestions how I might handle this?

**Comment by drarmstr** [comment, marked as answer]
React 18 Transitions are useful for this use case: https://recoiljs.org/docs/guides/transitions

**Reply by RobinClowers** [reply]
Sure enough, that solved my issue suspense fallbacks!

**Reply by RobinClowers** [reply]
Follow up here, I'm building a UI that relies on a few different api calls to get it's data, and I want to refresh all of it during a react transition. Is there some way to invalidate a whole tree of selectors, that's transition aware? I'm thinking of something like `useRecoilRefresher_UNSTABLE`, but that causes suspense fallbacks when I use it in this case.
